src/main/java/util/Parser.java

```diff
@@ -0,0 +1,172 @@
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * Helper function to remove the first command word
+     *
+     * @param command String obtained from scanner
+     * @return String without first command word
+     */
+    public static String removeCommandWord(String command) {
+        int firstSpaceIndex = command.indexOf(' ');
+
+        if (firstSpaceIndex == -1 || firstSpaceIndex == command.length() - 1) {
+            return "";
+        }
+        // Extract the substring starting from the position after the first space
+        return command.substring(firstSpaceIndex + 1).trim();
+    }
+
+    public static String getCommandWord(String command) {
+        return command.trim().toLowerCase().split(" ")[0];
+    }
+
+    /**
+     * Helper function to get the index from command
+     *
+     * @param command
+     * @return
+     */
+    public static int getIndex(String command) {
+        String[] commandList = command.trim().toLowerCase().split(" ");
+        if (commandList.length == 1) {
+            EpochMindException e = new EpochMindException("Thou hast forgotten to specify thine index");
+            System.out.println(e);
+            return -1;
+        } else {
+            try {
+                int index = parseInt(commandList[1]);
+                return index;
+            } catch (Exception e) {
+                System.out.println("Thou hast given an invalid index");
+            }
+            return -1;
+        }
+    }
+
+    /**
+     * Executes the command
+     *
+     * @param command User input
+     */
+    public String execute(String command) {
+        switch (getCommandWord(command)) {
+            case "bye":
+                return Ui.bye();
+            case "list":
+                return tasks.list();
+            case "mark":
+                return tasks.mark(getIndex(command));
+            case "unmark":
+                return tasks.unmark(getIndex(command));
+            case "todo":
+                return toDoParser(command);
+            case "deadline":
+                return deadlineParser(command);
+            case "event":
+                return eventParser(command);
+            case "delete":
+                return tasks.delete(getIndex(command));
+            case "overdue":
+                return tasks.overdue();
+            case "dueby":
+                return tasks.dueBy(command);
+            case "save":
+                return storage.save(tasks, removeCommandWord(command));
+            case "find":
+                return tasks.find(removeCommandWord(command));
+
+
+            default:
+                EpochMindException e = new EpochMindException("There is no such command in the Arcana of Knowledge");
+                return e.toString();
+        }
+    }
+
+    /**
+     * Parse command for todo creation
+     *
+     * @param command User Input
+     */
+    public String toDoParser(String command) {
```

Comment:
Maybe make the toDo a composite word, rather than 2 distinct words for camelCase? When I first saw the method, I thought it would convert this to a DoParser object of some kind, which confused me a lot.